### ISSUE TITLE
Randomly unchecks 'Extract torrent in-place'

### ISSUE THREAD

**Issue by mkulina** [cvarta/deluge-extractor#9]
Every few days, if I check preferences the 'extract torrent in-place' is unchecked causing all my downloads to not be extracted one they've finished. Happened on both deluge and binhex-delugevpn. 

Using unRAID. Let me know if you need any system specs or logs.
deluge 1.3.15

**Comment by aaomidi** [comment]
This issue exists for me too, infuriating.

**Comment by chrishuan9** [comment]
This has never happened to me before. @mkulina, @aaomidi  How can a checked preference in the settings uncheck itself ? I need more details to reproduce this issue

**Comment by spreadred** [comment]
I have never had this happen to me on either my Windows 10 or Ubuntu boxes. Are you sure the user you are running deluge as has the appropriate permissions to write to the config file directory?

**Comment by mkulina** [comment]
Yup. Because it will work for about 3-4 days. Then suddenly sonarr will say there is no file to import, and when I check the file. It hasn’t been extracted, which in turn I check my preferences and it has randomly been unchecked. 

The only plug ins I’m running are: MyScheduler, label, and simple extractor. If there’s anything else I can provide to assist you, please let me know. Running all from root user.

**Comment by darylm88** [comment]
This is happening to me as well. Running Deluge 1.3.15 on Win 7

**Comment by aaomidi** [comment]
My issue is exactly as @mkulina describes it to be.

**Comment by saitoh183** [comment]
same for me. I just loaded up my Deluge web and it was unchecked. Something to note is that in my thin client the options is unchecked and in the web client it was check. When i first opened the web client, a path was set and it was unchecked. But with the thin client, it was like nothing was configured.

My deluge daemon runs on a seed box and i connect over web and using my windows thin client.

**Comment by spreadred** [comment]
I stand corrected. I just got this bug as well. It also wiped out the "Extract to:" field. Deluge 1.3.15 on Ubuntu 16.04.4 LTS. Interestingly enough, looking in the config file itself still shows the options in enabled.

**Comment by chrishuan9** [comment]
This hasn’t happened to me - I will try to reproduce this issue, however any help like screenshots of the setting spähe and the configuration file  would be appreciated. 

To @all who experience this issue:
* Are the settings applied when it’s written correctly to the configuration file but not shown correctly in the configuration Settings page?
* In case the settings are not set in the settings UI, does reconfiguring in the Settings UI fix things ?

**Comment by spreadred** [comment]
@cvarta This screenshot represents the initial status of the UI versus the actual .conf:
![simple-extractor-conf](https://user-images.githubusercontent.com/6767624/40749394-db817cb0-6431-11e8-823a-a7a161e575ea.png)

I then checked the "Create torrent in place" and unchecked "Extract torrent in place". I then restarted the daemon and reconnected with the client. The simpleextractor.conf remained unchanged from its original state. 

Next I shutdown the daemon/client again and deleted the simpleextractor.conf~ file, but this did not resolve the issue either.

Lastly, I disabled the plugin in the UI, Shutdown the daemon/client and deleted the .conf file. I restarted the daemon/client and re-enabled the plugins selecting the desired options. I restarted it once more but the UI still refused to show the selected options however the brand new simpleextractor.conf had the appropriate settings.

It seems like beyond the initial selections made when enabling the plugin, the settings never get updated?

**Comment by chrishuan9** [comment]
@kaptainkommie , thank you for your clear description. I think I have a clear idea what causes the issue. I will make sure to fix this bug as soon as I am back home (business trip atm)

**Comment by ZataH** [comment]
Was this fixed? Because I still have this issue

**Comment by chrishuan9** [comment]
@ZataH, it hasn’t been fixed yet. I’m currently busy and can’t invest a lot of time in the development. I will fix this issue as soon as I find some time.

**Comment by aaomidi** [comment]
You said you had a clear idea of what was the cause. I could send in a PR if I you can give me some pointers?

**Comment by spreadred** [comment]
I haven't looked at this in a while, but I seem to recall I thought the problem lied within either the reading of the config file or the writing to the config file, or perhaps both?

**Comment by aaomidi** [comment]
Released a new version (forked) that doesn't fix this problem, but essentially always forces in place to be checked.

https://github.com/aaomidi/deluge-extractor/releases/tag/0.5

**Comment by anethema** [comment]
This is still a problem with the other check box 'create torrent name sub-folder'

It unchecks itself every several days. It breaks the flow for sonarr to plex so it is annoying heh.

**Comment by wipwar** [comment]
Still an issue on version 2.0.3-2-201906121747-ubuntu18.04.1

**Comment by Wardy125** [comment]
Yep, I'm getting this issue too.
Running in a docker container - it'll stick for a few hours, then until itself.
As @anethema said, it breaks the pipeline for some stuff coming in for radarr and sonarr.

**Comment by cjaustin1792** [comment]
I'm currently experiencing this issue.
Linux Ubuntu-2004-focal-64-minimal
Docker Deluge 2.0.3-2-201906121747

Checked config file and it is correct. View through WebUI and it is unchecked and files have been extracted into root folder

**Comment by HelSanYUL** [comment]
Still having this issue
Ubuntu Ubuntu 22.04.4 LTS
Deluge 2.1.1-4-05
Docker version 25.0.1, build 29cf629
